client/src/main/java/io/pravega/keycloak/client/KeycloakAuthzClient.java

```diff
@@ -104,6 +101,56 @@ public String getRPT() {
         return token.getToken();
     }
 
+    public void setHttpMaxRetries(int retries) {
+        httpMaxRetries = retries;
+    }
+
+    public void setHttpRetriesDelayMsecs(int delay) {
+        httpRetriesDelayMsecs = delay;
+    }
+
+    /**
+     * Executes a Keycloak client request with retries.
+     * HttpResponseExceptions may occur.  If the error code contained within is BAD_REQUEST,
+     * SC_UNAUTHORIZED or SC_FORBIDDEN, the request is not retried as this is an authentication error
+     * and will not be retried.
+     * Only HttpResponseExceptions with other error codes will be retried.
+     * Other exceptions will not be caught and will not be retried.
+     * @param f the client method to execute
+     * @param <T>
+     * @return
+     */
+    private <T> T withRetries(Supplier<T> f)  {
```

Comment:
Could we use a ready-made utility method like one of these here? There are a few in Pravega code below, that are used widely in Pravega and are well-tested.

https://github.com/pravega/pravega/blob/04e1825d3d3751da0bd7c5493d4b4c07adf7dc38/controller/src/main/java/io/pravega/controller/util/RetryHelper.java

Also, I think exponential backoff may be a more robust strategy for determining the retry interval, instead of regular interval retries, here. The above helper supports exponential backoff, as well.

Reply:
Great advice, yes, I reworked the code to use the `Retry` class and passed in my Predicate.